TestLogs: GET logs by limit: incorrect number of logs returned

<!--
************ STOP!! ************
If this issue identifies a security vulnerability, DO NOT submit it! Instead, contact
the Apache Traffic Control Security Team at security@trafficcontrol.apache.org and follow the
guidelines at https://www.apache.org/security/ regarding vulnerability disclosure.

- For *SUPPORT QUESTIONS*, use the Traffic Control slack (https://s.apache.org/atc-slack)
or Traffic Control mailing lists (https://trafficcontrol.apache.org/mailing_lists).
- Before submitting, please **SEARCH GITHUB** for a similar issue or PR.
-->

## I'm submitting a ...
<!-- delete all those that don't apply -->
<!--- security vulnerability (STOP!! - see above)-->
-  bug report

## Traffic Control components affected ...
<!-- delete all those that don't apply -->
-  Traffic Ops (API tests)

## Current behavior:
<!-- Describe how the bug manifests -->
The `TestLogs` API test fails when run by itself

## Expected behavior:
<!-- Describe what the behavior would be without the bug -->
All API tests should pass

## Minimal reproduction of the problem with instructions:
<!--
If the current behavior is a bug, please provide the *STEPS TO REPRODUCE* and
include the applicable TC version.
-->
In the `traffic_ops/testing/api` directory:
1. Start Traffic Ops

2. Configure `traffic-ops-test.conf`

3.
    ```shell script
    go test ./v3 -test.v -cfg=../conf/traffic-ops-test.conf -fixtures=tc-fixtures.json -test.run '^TestMain$|^TestLog$'
    ```

## Anything else:
<!--
e.g. stacktraces, related issues, suggestions how to fix (feel free to delete
this section)
-->
* Tested at c9ea0910ec
* Logs:
    ```go
    INFO: traffic_ops_test.go:71: 2020-10-07T09:30:12.297796111-06:00: Using Config values:
                   TO Config File:       ../conf/traffic-ops-test.conf
                   TO Fixtures:          tc-fixtures.json
                   TO URL:               https://localhost:6443
                   TO Session Timeout In Secs:  60
                   DB Server:            localhost
                   DB User:              traffic_ops
                   DB Name:              traffic_ops
                   DB Ssl:               false
                   UseIMS:               true
    === RUN   TestLogs
        logs_test.go:42: GET logs by limit: incorrect number of logs returned (0)
    --- FAIL: TestLogs (0.01s)
    FAIL
    FAIL	github.com/apache/trafficcontrol/traffic_ops/testing/api/v3	0.303s
    FAIL
    ```

<!--
Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.  See the NOTICE file
distributed with this work for additional information
regarding copyright ownership.  The ASF licenses this file
to you under the Apache License, Version 2.0 (the
"License"); you may not use this file except in compliance
with the License.  You may obtain a copy of the License at

    https://apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing,
software distributed under the License is distributed on an
"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, either express or implied.  See the License for the
specific language governing permissions and limitations
under the License.
-->

--- a/traffic_ops/testing/api/v1/logs_test.go
+++ b/traffic_ops/testing/api/v1/logs_test.go
@@ -20,7 +20,7 @@ import (
 )
 
 func TestLogs(t *testing.T) {
-	WithObjs(t, []TCObj{}, func() {
+	WithObjs(t, []TCObj{Roles, Tenants, Users}, func() { // Objs added to create logs when this test is run alone
 		GetTestLogs(t)
 		GetTestLogsByLimit(t)
 	})

--- a/traffic_ops/testing/api/v2/logs_test.go
+++ b/traffic_ops/testing/api/v2/logs_test.go
@@ -20,7 +20,7 @@ import (
 )
 
 func TestLogs(t *testing.T) {
-	WithObjs(t, []TCObj{}, func() {
+	WithObjs(t, []TCObj{Roles, Tenants, Users}, func() { // Objs added to create logs when this test is run alone
 		GetTestLogs(t)
 		GetTestLogsByLimit(t)
 	})

--- a/traffic_ops/testing/api/v3/logs_test.go
+++ b/traffic_ops/testing/api/v3/logs_test.go
@@ -20,7 +20,7 @@ import (
 )
 
 func TestLogs(t *testing.T) {
-	WithObjs(t, []TCObj{}, func() {
+	WithObjs(t, []TCObj{Roles, Tenants, Users}, func() { // Objs added to create logs when this test is run alone
 		GetTestLogs(t)
 		GetTestLogsByLimit(t)
 	})

